Framework/Algorithms/src/CopyInstrumentParameters.cpp

```diff
@@ -94,9 +96,33 @@ void CopyInstrumentParameters::exec() {
     // changed parameters
     m_receivingWorkspace->swapInstrumentParameters(targMap);
 
+    // Deal with parameters that are stored in Beamline::DetectorInfo
+    const auto &givingDetInfo = m_givingWorkspace->detectorInfo();
+    auto &receivingDetInfo = m_receivingWorkspace->mutableDetectorInfo();
+    try {
+      // If all detector IDs match a simple assignment should work.
+      receivingDetInfo = givingDetInfo;
+    } catch (std::runtime_error &) {
+      // Fallback for mismatch of detector IDs.
+      const auto &givingDetIDs = givingDetInfo.detectorIDs();
+      for (const auto detID : receivingDetInfo.detectorIDs()) {
+        // TODO Uncomment code and add handling for every field being added to
+        // Beamline::DetectorInfo.
+        // const auto receivingIndex = receivingDetInfo.indexOf(detID);
+        if (std::find(givingDetIDs.begin(), givingDetIDs.end(), detID) !=
+            givingDetIDs.end()) {
+          // const auto givingIndex = givingDetInfo.indexOf(detID);
+          // Copy values for all fields in Beamline::DetectorInfo
+        } else {
+          // Set default values for all fields in Beamline::DetectorInfo
```

Comment:
I see what you're doing here. Do we need to be stricter.

1. Checking that the InstrumentNames match?
1. Are we assuming that we can transfer parameter from newer to older instrument versions as well as the other way around? i.e are there any errors we want to flag up if we have detectorids missing? Maybe not?